### PR TITLE
Fe requests

### PR DESCRIPTION
**Pull request by mikez321**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling (Styling to new or existing files)
- [X] Testing
### Detailed Description
Added a few updates requested by the FE:
- A delete endpoint to remove plants from the garden
- Added a days till harvest field in the plant show
- Return a sad path for no plant found in plant search
### Why is this change required? What problem does it solve?
Adds functionality that will make the app more user friendly
### Was any previously written code changed?
Nope

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -267,6 +267,21 @@ Response:
 }
 ```
 
+### DELETE Endpoints:
```
Comment (kelshadarby) [inline README.md:270]: Nice work on this! I think we should change the watering endpoint from a post to a put or a patch as well

File: app/api/routes.py [hunk 2]
```diff
@@ -61,15 +61,21 @@ def random_plant():
...
-    results = []
 
-    for plant in plants:
-        obj = {
-            'id': plant.id,
-            'plant_type': plant.plant_type,
-            'plant_image': plant.image,
+    if len(plants) == 0:
+        results = {
+            'image': 'error image',
```
Comment (kelshadarby) [inline app/api/routes.py:67]: Nice sad path consideration here! We'll have to input a picture here later as well

File: README.md (hunk 1 above)
Comment (mikez321) [reply README.md:270]: I'll make a card for it and we can do that as a 'clean up' item.  Good catch!

File: app/api/routes.py [hunk 3]
```diff
@@ -147,18 +161,39 @@ def update_watering():
...
+        request.method == 'DELETE'
+        gardenplant = GardenPlant.query.get_or_404(id)
+        result = {
+            'gardenplant_id': gardenplant.id,
+            'plant_name': gardenplant.plant_name,
+        }
+        db.session.delete(gardenplant)
+        db.session.commit()
+        response = jsonify(result)
+        response.status_code = 202
```
Comment (kelshadarby) [inline app/api/routes.py:198]: Very clean and easy to read

**Pull request merged**